packages/uninstaller/src/gcp.ts

```diff
@@ -28,33 +28,39 @@ import { destroyDNS } from "@opstrace/dns";
 import { log, getBucketName } from "@opstrace/utils";
 
 import { destroyConfig } from "./index";
-
-export function* destroyGCPInfra() {
+import { Task } from "redux-saga";
+
+export function* destroyGCPInfra(): Generator<
+  JoinEffect | CallEffect | ForkEffect | Generator<ForkEffect, Task[], Task>,
+  void,
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  any
+> {
   log.info(`Ensure cert-manager service account deletion`);
   yield call(ensureServiceAccountDoesNotExist, {
     name: `${destroyConfig.clusterName}-cert-manager`,
-    projectId: destroyConfig.gcpProjectID!,
+    projectId: destroyConfig.gcpProjectID ?? "",
```

Comment:
@juliachvyrova what would be the typescript best practice to ensure `destroyGCPInfra` would only be called if these (and other) properties are set correctly?

Reply:
TS doesn't do runtime type checking, so you have to do that yourself. Best way would be to assert the properties you expect are indeed there, as you'd expect them and throw and Error if not.

Only the properties that can __change__ at runtime (i.e. user provided config for example) need to be checked at runtime, since the rest will benefit from static analysis during compilation. If you don't want to do your own simple typecheck, some libraries that do runtime checking:

https://github.com/jquense/yup
https://github.com/gcanti/io-ts

There are many.

In the case here, it is probably safer to add this before the call to `ensureServiceAccountDoesNotExist`:
```
if (!destroyConfig.gcpProjectID) {
  throw Error("gcp project id not specified in credentials");
}
```

Reply:
@MatApple Good idea! I thought about checking it, because in some places error will be raised anyway ( like when we get path from env variables and then try to get file by that path ), but having explicit error is always better for future debugging. :)

Reply:
> but having explicit error is always better for future debugging. :)

:100:

Reply:
I was thinking it should be possible to check the config object once (somewhere) and then we wouldn't have to rely on `obj!.someProperty` or `obj?.someProperty` type. I guess that's what `yup` is for but I was expecting not to do `!` or `?` afterwards.